### PR TITLE
Reverting service state changes, return changes in pkg state

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Reverting changes in PR #60150. Updating installed and removed functions to return changes when test=True.

### What issues does this PR fix or reference?
Fixes: #60995 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by Ch3LL — APPROVED** [review]

**Comment by meaksh** [comment]
Hey @garethgreenaway, we just noticed some of the changes introduced here are a bit confusing and causing a kind of inconsistency, maybe a bug.

Why the returned `changes` contains now a `installed` key when `test=True`, and only for package installations?

According to Salt state return data documentation [1], for `changes`, _each thing changed should be a key, with its value being another dict with keys called "old" and "new" containing the old/new values_:  https://docs.saltproject.io/en/latest/ref/states/writing.html#return-data

This PR is changing this described behavior when `test=True` (and only for package installations), and it now returns:

```
changes: {
  'installed' : {
    'adobe-sourcecodepro-fonts' : {
      'new' : '2.030-1.30'
      'old' : ''
    }
  }
}
```

This seems wrong, as I'd expect this instead:

```
changes: {
  'adobe-sourcecodepro-fonts' : {
    'new' : '2.030-1.30'
    'old' : ''
  }
}
```

As mentioned, for package removal, with `test=True` the output looks correct as expected: https://github.com/saltstack/salt/pull/61019/files#diff-9a4a8768915a83c4c1dda97ec682168ef1849c5896a5f2e0397097021204ffd0R2963 

Is this a bug or was this change done on purpose? What is the desired behavior here? 

Thanks for clarifying!

**Comment by garethgreenaway** [comment]
@meaksh The change was done on purpose, we needed to be able to see what would be changed when `test=True` was set.  Previously the changes were empty in that case, which when this state was paired with the service state in a `watch` setup, would cause the output to indicate that the service would restart every time.  I'd have to dive in more and see why `installed` is in the changes now, we may been able to just have it by the normal format.

**Comment by damon-atkins** [comment]
These should be separate PRs,   a role back and an enhancement in one PR?

**Comment by damon-atkins** [comment]
@garethgreenaway  why does the output format need to change for test=true. I would expect it to  output the same information as if it was running for real, and skip calling exec install and assume it worked.    Possible return 'latest' as the version if it can not calculate the version.       Might be worth looking at the Windows Code with the state which does a calculation on version (its on purpose it does this).

File: salt/states/pkg.py [hunk 1]
```diff
@@ -1822,12 +1822,17 @@ def installed(
         )
 
     comment = []
+    changes = {"installed": {}}
```
Comment (damon-atkins) [inline salt/states/pkg.py:1825]: Prefixing the results with "installed" ?

**Comment by Ch3LL** [comment]
@garethgreenaway submitted this PR https://github.com/saltstack/salt/pull/62455 with the fix. Can you review @meaksh please?

**Comment by meaksh** [comment]
@Ch3LL @garethgreenaway great, looks good, thanks :+1:

File: salt/states/pkg.py (hunk 1 above)
Comment (damon-atkins) [reply salt/states/pkg.py:1825]: @garethgreenaway thanks for removing the installed prefix in #62455

**Pull request merged**